app/controllers/provider_interface/start_page_controller.rb

```diff
@@ -1,14 +1,19 @@
 module ProviderInterface
   class StartPageController < ProviderInterfaceController
+    COURSES_COUNT_CACHE_KEY = 'provider_interface_courses_count'.freeze
+    PROVIDERS_COUNT_CACHE_KEY = 'provider_interface_providers_count'.freeze
     before_action :redirect_authenticated_provider_user
     skip_before_action :authenticate_provider_user!
 
     layout 'base'
 
     def show
-      courses = Course.current_cycle.includes(:provider).open_on_apply
-      @course_count = courses.count
-      @provider_count = courses.map(&:provider).uniq.count
+      @course_count = Rails.cache.fetch(COURSES_COUNT_CACHE_KEY, expires_in: 24.hours) do
+        Course.current_cycle.open_on_apply.count
```

Comment:
I'd lean on Rails helpers: generate keys here, and just cache in the view? As done here:

https://github.com/DFE-Digital/apply-for-teacher-training/blob/e010ceb844c3f686c28d5a329307214afcbc7ded/app/views/candidate_interface/course_choices/provider_selection/new.html.erb#L13-L23

and use one cache for both numbers, which occur side by side in the markup

Reply:
This is a good idea, but we'll need to make sure the count queries are triggered by the view. I'm guessing the hard work in this other example is triggered by `.available_providers`, which is why caching the view is effective.

If we leave the calculation of @provider_count and @course_count in the controller and simply reference them in the view, I don't think we'll get much benefit, as we are trying to avoid the queries.

Reply:
@mnacos is correct, I've moved the parsing into the view. Tested this locally and seems to do the job, let me know your thoughts @vigneshmsft @duncanjbrown

Reply:
Yeah — I assumed the queries were relations in the controller but realised in the view, but of course that's not necessarily so. I think what's there now is good tho!